aes/src/lib.rs

```diff
@@ -123,12 +130,26 @@
 )]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 #![warn(missing_docs, rust_2018_idioms)]
+#![cfg_attr(
+    all(
+        any(target_arch = "riscv32", target_arch = "riscv64"),
+        target_feature = "zknd",
+        target_feature = "zkne"
+    ),
+    feature(riscv_ext_intrinsics)
```

Comment:
Please use the same approach as in the `sha2` crate, i.e. introduce a configuration flag which unconditionally enables the scalar backend, relies on Nightly, and requires user to enable the target features. It's not correct to switch to Nightly-only based on enabled target features.